### PR TITLE
Update links in framework/data folder

### PR DESCRIPTION
**Pull request by gewarren**
- replace redirected links
- replace site-relative links with file-relative links

Contributes to #15560

### CODE REVIEW

**Comment by gewarren** [comment]
@stevestein Are you okay with these changes?

File: docs/framework/data/adonet/ef/language-reference/queries-in-linq-to-entities.md [hunk 1]
```diff
@@ -62,6 +62,6 @@ A query is an expression that retrieves data from a data source. Queries are usu
 ## See also
 
 - [LINQ to Entities](linq-to-entities.md)
-- [Getting Started with LINQ in C#](../../../../../csharp/programming-guide/concepts/linq/getting-started-with-linq.md)
+- [Getting Started with LINQ in C#](/dotnet/csharp/programming-guide/concepts/linq/)
```
Comment (mairaw) [inline docs/framework/data/adonet/ef/language-reference/queries-in-linq-to-entities.md]:
```suggestion
- [Getting Started with LINQ in C#](../../../../../csharp/programming-guide/concepts/linq/index.md)
```

File: docs/framework/data/adonet/queries-in-linq-to-dataset.md [hunk 2]
```diff
@@ -57,5 +57,5 @@ A query is an expression that retrieves data from a data source. Queries are usu
 
 - [Programming Guide](programming-guide-linq-to-dataset.md)
 - [Querying DataSets](querying-datasets-linq-to-dataset.md)
-- [Getting Started with LINQ in C#](../../../csharp/programming-guide/concepts/linq/getting-started-with-linq.md)
+- [Getting Started with LINQ in C#](/dotnet/csharp/programming-guide/concepts/linq/)
```
Comment (mairaw) [inline docs/framework/data/adonet/queries-in-linq-to-dataset.md]:
```suggestion
- [Getting Started with LINQ in C#](../../../csharp/programming-guide/concepts/linq/index.md)
```

File: docs/framework/data/adonet/secure-data-access.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ To write secure ADO.NET code, you have to understand the security mechanisms ava
   
 |Resource|Description|  
 |--------------|-----------------|  
-|[Exception Handling Fundamentals](../../../standard/exceptions/exception-handling-fundamentals.md)|Describes the basic forms of try/catch/finally structured exception handling.|  
+|[Exception Handling Fundamentals](/dotnet/standard/exceptions/)|Describes the basic forms of try/catch/finally structured exception handling.|
```
Comment (mairaw) [inline docs/framework/data/adonet/secure-data-access.md]:
```suggestion
|[Handling and throwing exceptions in .NET](../../../standard/exceptions/index.md)|Describes the basic forms of try/catch/finally structured exception handling.|  
```

**Review by mairaw — APPROVED** [review]
LGTM @gewarren. Left a few suggestions for you to consider.

**Comment by gewarren** [comment]
Thanks for the review Maira!

**Pull request merged**